### DISCUSSION TITLE
[Feature] Selecting multiple tags

### DISCUSSION THREAD

**Discussion by ilyaChuk** [immich-app/immich#12165] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I don't understand why it's not there at the beginning. This is the essence of the tag system. 
I tag photos with the "cat" tag. I tag photos with the tag "dog". I tag photos with the "cute" tag. Why can't I select the "cat" and "cute" tags on the page `/tags` **at the same time**? I can **only** choose **one** thing.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
What is the use case of selecting two tags in the viewer? Also can you provide example from other software that handles such use case?

**Reply by ilyaChuk** [reply]
the very point of using tags is to then select intersections. for example, the intersection of the tags "friends" and "family". the intersection of the tags "Italy" and "cafe". the intersection of the "report" and "signed" tags. 
After all, there is such a thing almost everywhere where there are tags. In Evernote, Nation and, as far as I understand, in Adobe Lightroom.

**Reply by ilyaChuk** [reply]
although it would be even more useful to choose tag intersection or union

**Reply by alextran1502** [reply]
What would be more fitting to be included in the search feature I think.

**Reply by danieldietzler** [reply]
We absolutely plan on letting users select multiple tags to filter by. I am *very* sure that's already in the existing feature request discussion about tags though

**Reply by danieldietzler** [reply]
This is just the first iteration of tags, to get something out of the door. We plan on adding a bunch more in later iterations, don't worry :D

**Comment by 0815IDIOT** [comment]
I would like to add a though to this discussion: Additionally to search by multiple tags, I would also love to be able to have a "not" operator when searching for multiple tags. E.g. photos that are "dogs" and not "cute". 
Looking forward to what going to come!

**Comment by Misfi** [comment]
Upvote for multi-tag search.
What I would also love to see:
1) Filters based on tags -> save your most common tag combinations to be re-used
2) AND, OR, NOT operators
3) Using the multi-tags for albums to automatically include all matching photos

**Comment by MD77MD** [comment]
my input on this Feature which i think apply for both tags and albums. 

Currently we can only add an asset to multiple albums one at a time... this takes time especially if your adding many assets... because we have to do the hunt and reselect all the assets i want to add for every album i want to add them to.

i wish we can just add assets to multiple albums at once for example like this
![SmartSelect_20241119_001017_Vivaldi](https://github.com/user-attachments/assets/ad0d47b3-22bb-4b14-9d25-506bdc121410)

where ticked album would rise to the top... meaning: 

for example we have album (a,b,c,d,e) show in list view
when i tick/select album (c) and (d), both dynamic will rise to the top of the list. under them will be the remaining unticked albums which are (a,b,e)

this way we can review selected album at the top of the list for any adjustments before applying the tags 

this can be applied for both tags and albums

**Comment by Rebellce** [comment]
我认为可以搜索标签是很实用的功能，它可以快速帮助我整理包含某些共同标签的照片到相册中，而无需在海量的照片中翻找 : D
I think search tags are a very useful feature; they can quickly help me organize photos with certain common tags into an album without having to rummage through a vast number of photos.

**Comment by Alphaelectric** [comment]
I just installed my server and was surprised that it only allows searching for one tag. As others pointed out tags create intersections to enable a multitude of search results. Is this complex to implement? DB wise it's just adding another select after the first one. Multiple selects are standard procedure for sorting large collections. +1 for this feature.

**Comment by cammurray** [comment]
This feature would be extremely useful, especially if you're in to photography like myself. I would like to be able to "refine" my search in the tag view, e.g start off with landscape, then potentially click "trees", and granually be able to find what I need.

I guess what would be second to this, or alternative, is if supported human language search. E.g "show me landscape photos that have trees and no houses" or something.
